app/channel.go

```diff
@@ -766,7 +766,10 @@ func (a *App) UpdateChannelMemberNotifyProps(data map[string]string, channelId s
 }
 
 func (a *App) DeleteChannel(channel *model.Channel, userId string) *model.AppError {
-	ihc := a.Srv.Store.Webhook().GetIncomingByChannel(channel.Id)
+	incomingHooks, err := a.Srv.Store.Webhook().GetIncomingByChannel(channel.Id)
```

Comment:
The point here is to use the async version explicitly, here is an example extracted from the example PR:

```go
	hchan := make(chan store.StoreResult, 1)
	go func() {
		webhook, err := a.Srv.Store.Webhook().GetIncoming(hookId, true)
		hchan <- store.StoreResult{Data: webhook, Err: err}
		close(hchan)
	}()
```